overlord/snapstate/snapmgr.go

```diff
@@ -511,6 +530,32 @@ func (m *SnapManager) ensureRefreshes() error {
 	return err
 }
 
+// ensureCatalogRefresh ensures that we refresh the cataloge
+// data periodically
+func (m *SnapManager) ensureCatalogRefresh() error {
+	// sneakily don't do anything if in testing
```

Comment:
We could move the mkdir here so that it stays operational even if someone just goes and does `rm -rf` on the thing after snapd started.